var/spack/repos/builtin/packages/evtgen/package.py

```diff
@@ -27,6 +27,7 @@ class Evtgen(CMakePackage):
     variant('hepmc3', default=False, description='Link with hepmc3 (instead of hepmc)')
 
     patch("g2c.patch", when='@01.07.00')
+    patch("evtgen-2.0.0.patch", when='@02.00.00')
```

Comment:
Does this still work with pythia < 8.304? If not the `when` should be updated

Reply:
Ah, apologies, I didn't see the `#if` above, that's fine then.

Reply:
Wouldn't it be easier (and safer, and avoid the extra `patch` for versions that natively work) to change this to `when='@02.00.00 ^pythia@8.304:'` and have your file simply replace `Pythia8::ParticleDataEntry*` with `Pythia8::ParticleDataEntryPtr`?

Reply:
@sethrj what do you mean by "safer"? Anyway, this is only needed for 02.00.00, the patch was accepted upstream and included in 02.01.00

Reply:
Safer by the time-honored principle "if it ain't broke don't fix it" 😉 . I still think it would be better add the `^pythia@8.304:` to avoid the unnecessary patch step and longer spec name, even though this will be increasinly irrelevant (since both versions will be out of date).